apps/frontend/src/pages/CreateListing.tsx

```diff
@@ -1,5 +1,281 @@
+import {
+  Box,
+  Button,
+  Card,
+  CardContent,
+  Container,
+  FormControl,
+  Grid,
+  TextField,
+  Typography,
+} from "@mui/material";
+import { useCallback, useEffect, useState } from "react";
+import _axios_instance from "../_axios_instance.tsx";
+import { colors } from "../styles/colors.tsx";
+import MultiFileUpload from "../extra_components/MultiFileUpload.tsx";
+import Carousel from "../extra_components/Carousel.tsx";
+
+interface URLObject {
+  url: string;
+}
+interface LocationObject {
+  latitude: number;
+  longitude: number;
+}
+
+interface ListingObject {
+  title: string;
+  description: string;
+  price: number;
+  location: LocationObject;
+  images: URLObject[];
+}
+
+interface NewListingObject {
+  listing: ListingObject;
+}
+
 const CreateListing = () => {
-  return <h1>CreateListing</h1>;
+  const [images, setImages] = useState<string[]>([]);
+
+  // This newListingObject bundles all the listing data for upload to the server
+  const [newListingObject, setNewListingObject] = useState<NewListingObject>({
+    listing: {
+      title: "",
+      description: "",
+      price: 0,
+      location: {
+        latitude: 0.0,
+        longitude: 0.0,
+      },
+      images: [],
+    },
+  });
+
+  // Gets the user location, and adds it to the listing object
+  const getUserLocation = useCallback(async () => {
+    try {
+      navigator.geolocation.getCurrentPosition((position) => {
+        newListingObject.listing.location.longitude = position.coords.longitude;
+        newListingObject.listing.location.latitude = position.coords.latitude;
+      });
+      return true;
+    } catch (error) {
+      return false;
+    }
+  }, [newListingObject.listing.location]);
+
+  // Updates and sends the newListingObject, to the server via post under /api/listing
+  const sendPostToCreateListing = async () => {
+    // In order to make sure that the images are retrieved before submitting
+    const successImages = await uploadImages();
+    let successLocation: boolean = true;
+    if (newListingObject.listing.location.latitude !== 0) {
+      successLocation = false;
+      getUserLocation().then(() => {
```

Comment:
I think this might set successLocation to true regardless of the return value from getUserLocation, has this been tested?

Reply:
I've done quite a few manual tests, but I stopped when I realized that some browsers straight up block geolocation. I was thinking of changing this to instead default the location to the center of campus and then the other components can do the same if getting location fails.

It's not the most elegant solution, but there isn't much we can do without manual entry of a location